### PR TITLE
Add traits for better documentation

### PR DESCRIPTION
**Pull request by kpob**
## Summary by CodeRabbit

- **New Features**
    - Enhanced host-contract interaction capabilities with new traits and structs for improved contract handling.
    - Introduced functionality for hosting smart contracts, including deployment, event handling, and interaction with the host environment.

- **Refactor**
    - Reorganized module imports related to the host environment across several files for clarity and efficiency.
    - Updated various contracts and features to utilize new host interaction capabilities for deployment and initialization.
    - Restructured approach to importing and utilizing core functionalities, particularly in `odra-casper` and `odra-macros` directories.

- **Documentation**
    - Minor correction in code comment regarding proper usage of a directive.

- **Chores**
    - Adjustments in import statements across multiple directories to align with the enhanced host interaction capabilities.

### CODE REVIEW

File: examples/src/contracts/tlw.rs [hunk 1]
```diff
@@ -101,14 +101,22 @@ pub struct Withdrawal {
 mod test {
     use super::*;
     use crate::contracts::tlw::{Deposit, Withdrawal};
-    use odra::Address;
+    use odra::{
+        host::{Deployer, HostRef},
+        Address
+    };
```
[the remark this answers is not in the document]
Comment (kpob) [reply examples/src/contracts/tlw.rs:107]: To deploy a contract we call Deployer::deploy which is implemented by a contract HostRef. If we remove it form all the places you pointed, the code would not compile

File: core/src/host.rs [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+    }
+
+    fn into_runtime_args(self) -> Option<RuntimeArgs> {
+        Some(RuntimeArgs::new())
+    }
+}
+
+/// An implementation of [InitArgs]. Should be used when the contract
+/// does not have the constructor.
+pub struct NoneInitArgs;
```
Comment (zie1ony) [inline core/src/host.rs]:
```suggestion
pub struct NoInit;
```
I think `NoInit` is better as it suggests that there is no init function.

File: core/src/host.rs [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+    ///
+    /// If the `expected_ident` does not match the contract ident, the method returns `false`.
+    fn validate(expected_ident: &str) -> bool;
+    /// Converts the init args into a [RuntimeArgs] instance.
+    fn into_runtime_args(self) -> Option<RuntimeArgs>;
+}
+
+/// Default implementation of [InitArgs]. Should be used when the contract
+/// has the constructor but does not require any initialization arguments.
+pub struct EmptyInitArgs;
```
Comment (zie1ony) [inline core/src/host.rs]:
```suggestion
pub struct NoArgs;
```

File: odra-macros/tests/test.rs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+pub mod odra {
```
Comment (zie1ony) [inline odra-macros/tests/test.rs:1]: You just discovered new ways of testing Odra code with mocks. Nice :)

**Review by zie1ony — APPROVED** [review]

**Pull request merged**